### PR TITLE
Executor, Model and completed Monitor

### PR DESCRIPTION
**Pull request by javelinjs**
1. Executor, without DataParallelExecutorManager (**NOT tested**)
2. Model, only static (object) functions (**NOT tested**)
3. Monitor.scala completed. MonitorCallback jni implementation depends on a minor patch in mxnet::Executor::SetMonitorCallback, I'll PR it later. (**NOT tested**)
4. Add norm function in NDArray, which broke the test on travis linux. I fixed it in a temporary way. If anything goes wrong on your guys own laptop, please let me know. (**Tested**)

### CODE REVIEW

File: scala-package/core/src/main/scala/ml/dmlc/mxnet/Monitor.scala [hunk 1]
```diff
@@ -18,97 +20,96 @@ class Monitor(protected val interval: Int, protected var statFunc: (NDArray) =>
...
    * End collecting and print results
    */
-  def tocPrint = {
-    val res = toc
-    res.foreach { re =>
-      val (n, k, v) = re
-      logger.info(s"Batch: ${n} ${k} ${v}")
+  def tocPrint(): Unit = {
+    val res = toc()
+    res.foreach { case (n, k, v) =>
```
Comment (terrytangyuan) [inline scala-package/core/src/main/scala/ml/dmlc/mxnet/Monitor.scala]: Just curious: any reasons of using `case` here but `val (n, k, v)` in `toc()`?

File: scala-package/core/src/main/scala/ml/dmlc/mxnet/Executor.scala [hunk 2]
```diff
@@ -1,11 +1,231 @@
...
+  /**
+   * Get input slice from the input shape.
+   * @param batchSize The number of samples in a mini-batch.
+   * @param workLoadList The list of work load for different devices, in the same order as ctx
+   * @return The split slices to get a specific slice.
+   * @throws IllegalArgumentException If there are two many splits such that some slice can be empty.
+   */
+  private def splitInputSlice[@specialized(Int, Float, Double) V]
+                              (batchSize: Int, workLoadList: Array[V])
+                              (implicit num: Numeric[V]): Array[(Int, Int)] = {
```
Comment (terrytangyuan) [inline scala-package/core/src/main/scala/ml/dmlc/mxnet/Executor.scala]: Any particular reason why we need `implicit num: Numeric[V]` here? Wouldn't `Int, Float, Double` be enough type options for elements in `workLoadList`. Could you clarify? :-)

File: scala-package/core/src/main/scala/ml/dmlc/mxnet/Monitor.scala (hunk 1 above)
Comment (javelinjs) [inline scala-package/core/src/main/scala/ml/dmlc/mxnet/Monitor.scala]: nothing special, the way you wrote is cool.

File: scala-package/core/src/main/scala/ml/dmlc/mxnet/Executor.scala (hunk 2 above)
Comment (javelinjs) [inline scala-package/core/src/main/scala/ml/dmlc/mxnet/Executor.scala]: workLoadList.sum needs it

**Pull request merged**